src/datasources/cache/cache.first.data.source.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { faker } from '@faker-js/faker';
+import { NetworkResponseError } from '../network/entities/network.error.entity';
+import { mockNetworkService } from '../network/__tests__/test.network.module';
+import { CacheFirstDataSource } from './cache.first.data.source';
+import { ICacheService } from './cache.service.interface';
+import { CacheDir } from './entities/cache-dir.entity';
+import { fakeCacheService } from './__tests__/test.cache.module';
+
+describe('CacheFirstDataSource', () => {
+  let cacheFirstDataSource: CacheFirstDataSource;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
```

Comment:
Why not create a `FakeCacheService` here instead of using a global one?

Actually this can lead to bugs and we should change all the instances in the future. Access to global components should be done via the module and not via an exported const.

Reply:
Yes, I agree having a common instance exported by a constant to multiple places is not very convenient. This affects many test classes, I think this could be tackled separately if you agree 🙂

Reply:
Yes, I agree having a common instance exported by a constant to multiple places is not very convenient. This affects many test classes, I think this could be tackled separately if you agree 🙂

Reply:
> Yes, I agree having a common instance exported by a constant to multiple places is not very convenient. This affects many test classes, I think this could be tackled separately if you agree 🙂

There are two scenarios in most of our tests:
- Reference the `fakeCacheService` via a global reference _because_ that's one (not so good) way to get the reference to the cache that will be used by the app module
- Reference to the `fakeCacheService` directly – use the reference to build the target class being tested – this is the use case of this test and we should address it here. 

If we start getting the cache instance via the module it won't change this test because there are no modules involved.

Reply:
Makes sense, thanks, changing it in [7e82cff](https://github.com/safe-global/safe-client-gateway-nest/pull/295/commits/7e82cff13ee7845879c4cb33aee1317a83082372)